src/Cabal/Plan.hs

```diff
@@ -179,6 +179,13 @@ data Unit = Unit
        -- ^ Source of the package
        --
        -- @since 0.5.0.0 (TODO)
+    , uOptional :: !(Maybe Bool)
+       -- ^ Is this unit considered optional? Optional targets are not built by
+       -- v2-build unless specifically asked for. For example @cabal v2-build
+       -- all@ will not build optional targets.
+       --
+       -- @since 0.5.0.0 (TODO). Available in @plan.json@ starting with
```

Comment:
...related to the previous comment: If the type of `uOptional` is to remain `Maybe Bool`, then we need a fixup traversal in the top-level parsing function, which fixes up `Nothing` to `Just False` if the version field has the respective version.

If `uOptional` is to become `Bool`, we need a top-level "feature" predicate which allows to query whether the `uOptional` field can be trusted.